### PR TITLE
⚠️ Adds missing category field to VSphereVM CRD

### LINKED ISSUE #1799: VSphereVM CRD is missing the `cluster-api` category label
/kind bug

**What steps did you take and what happened:**
All the CAPV CRDs have their `category` set to `cluster-api`. But the `VSphereVM` CRD is missing the category field altogether. 

**What did you expect to happen:**
The `VSphereVM` CRD has the category field set.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-vsphere version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by srm09**
**What this PR does / why we need it**:
Adds missing category to the VSphereVM CRD.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1799 

**Special notes for your reviewer**:
n/a

**Release note**:
```release-note
Adds missing category to the VSphereVM CRD.
```

### CODE REVIEW

File: apis/v1beta1/vspherevm_types.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ type VSphereVMStatus struct {
 }
 
 // +kubebuilder:object:root=true
-// +kubebuilder:resource:path=vspherevms,scope=Namespaced
+// +kubebuilder:resource:path=vspherevms,scope=Namespaced,categories=cluster-api
```
Comment (adam-jian-zhang) [inline apis/v1beta1/vspherevm_types.go:145]: What's the consquence if missing?

**Review by adam-jian-zhang — APPROVED** [review]

File: apis/v1beta1/vspherevm_types.go (hunk 1 above)
Comment (srm09) [reply apis/v1beta1/vspherevm_types.go:145]: Querying all the resources belonging to the cluster-api category would result in the VsphereVM CRs not show up. Mainly would affect any diagnostic bundles, which would used the cluster-api category to pull all the resources belonging to that category.

**Comment by sbueringer** [comment]
Seems unrelated to the change. 

Maybe also some sort of controller-tools/controller-gen version mismatch between local and prow

**Comment by srm09** [comment]
/override pull-cluster-api-provider-vsphere-verify-gen 
Because of a flake which is being tracked in #1757

**Comment by srm09** [comment]
/override pull-cluster-api-provider-vsphere-verify-crds
Similar reason as above

**Pull request merged**